rcl/test/CMakeLists.txt

```diff
@@ -1,117 +1,86 @@
 find_package(ament_cmake_gtest REQUIRED)
+find_package(std_msgs REQUIRED)
+
+include(rcl_add_custom_gtest.cmake)
 
 set(extra_test_libraries)
-set(extra_memory_tools_env PHONY=)  # Use a phony env var so there is always at least one.
-# rcl_lib_dir is passed as APPEND_LIBRARY_DIRS for each ament_add_gtest call so
-# the librcl that they link against is on the library path.
-# This is especially improtant on Windows.
-set(rcl_lib_dir $<TARGET_FILE_DIR:${PROJECT_NAME}>)
-ament_find_gtest()  # For GTEST_LIBRARIES
-if(APPLE)
-  add_library(${PROJECT_NAME}_memory_tools_interpose SHARED memory_tools_osx_interpose.cpp)
-  target_link_libraries(${PROJECT_NAME}_memory_tools_interpose ${GTEST_LIBRARIES})
-  set_target_properties(${PROJECT_NAME}_memory_tools_interpose
-    PROPERTIES COMPILE_FLAGS "-std=c++11")
-  list(APPEND extra_test_libraries ${PROJECT_NAME}_memory_tools_interpose)
-  list(APPEND extra_memory_tools_env
-    DYLD_INSERT_LIBRARIES=$<TARGET_FILE:${PROJECT_NAME}_memory_tools_interpose>)
-endif()
-add_library(${PROJECT_NAME}_memory_tools SHARED memory_tools.cpp)
-if(NOT WIN32)
-  set_target_properties(${PROJECT_NAME}_memory_tools PROPERTIES COMPILE_FLAGS "-std=c++11")
-endif()
-if(UNIX AND NOT APPLE)
-  list(APPEND extra_test_libraries dl)
-  list(APPEND extra_memory_tools_env DL_PRELOAD=$<TARGET_FILE:${PROJECT_NAME}_memory_tools>)
-endif()
-target_link_libraries(${PROJECT_NAME}_memory_tools ${extra_test_libraries})
-target_compile_definitions(${PROJECT_NAME}_memory_tools
-  PRIVATE "RCL_MEMORY_TOOLS_BUILDING_DLL")
-list(APPEND extra_test_libraries ${PROJECT_NAME}_memory_tools)
+set(extra_test_env PHONY=)  # Use a phony env var so there is always at least one.
```

Comment:
The disadvantage of that approach is that it isn't clear at the place of calling that you're passing anything for `ENV`.

After playing around with it, CMake seems perfectly happy to take it without it being set. Can you see anything wrong with b47ee17? It works as expected on my machine. I'll start a new Linux CI just to double check: http://ci.ros2.org/job/ci_linux/948/